File: src/models/RestaurantKind.ts
export enum RestaurantKind {
  Cafeteria,
  Restaurant,
  ApprovedRestaurant,
  ManagedRestaurant,
  AdministrativeRestaurant,
  CoffeeCorner,
  Brewery,
  FoodTruck,
  SelfService,
  Kiosk,
  Pizzeria,
  GroceryStore,
  Scooter,
  CrousAndGo,
  SandwichShop
}

export const translateRestaurantKindFromAPI = (api: string): RestaurantKind => {
  switch (api) {
    case "Cafétéria":
      return RestaurantKind.Cafeteria;
    case "Restaurant":
      return RestaurantKind.Restaurant;
    case "Restaurant agréé":
      return RestaurantKind.ApprovedRestaurant;
    case "Restaurant géré":
      return RestaurantKind.ManagedRestaurant;
    case "Coffee Corner":
      return RestaurantKind.CoffeeCorner;
    case "Brasserie":
      return RestaurantKind.Brewery;
    case "Foodtruck":
      return RestaurantKind.FoodTruck;
    case "Restaurant administratif":
      return RestaurantKind.AdministrativeRestaurant;
    case "Libre-service":
      return RestaurantKind.SelfService;
    case "Kiosque":
      return RestaurantKind.Kiosk;
    case "Pizzéria":
      return RestaurantKind.Pizzeria;
    case "épicerie":
      return RestaurantKind.GroceryStore;
    case "Triporteur":
      return RestaurantKind.Scooter;
    case "crous and go":
      return RestaurantKind.CrousAndGo;
    case "Sandwicherie":
      return RestaurantKind.SandwichShop;
    default:
      throw new Error(`Unknown restaurant type: ${api}`);
  }
};

File: library/src/commonMain/kotlin/ink/literate/crowous/models/restaurant-kind.kt
package ink.literate.crowous.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class RestaurantKind {
    @SerialName("CAFETERIA")
    CAFETERIA,
    @SerialName("RESTAURANT")
    RESTAURANT,
    @SerialName("APPROVED_RESTAURANT")
    APPROVED_RESTAURANT,
    @SerialName("MANAGED_RESTAURANT")
    MANAGED_RESTAURANT,
    @SerialName("ADMINISTRATIVE_RESTAURANT")
    ADMINISTRATIVE_RESTAURANT,
    @SerialName("COFFEE_CORNER")
    COFFEE_CORNER,
    @SerialName("BREWERY")
    BREWERY,
    @SerialName("FOOD_TRUCK")
    FOOD_TRUCK,
    @SerialName("SELF_SERVICE")
    SELF_SERVICE,
    @SerialName("KIOSK")
    KIOSK,
    @SerialName("PIZZERIA")
    PIZZERIA,
    @SerialName("GROCERY_STORE")
    GROCERY_STORE,
    @SerialName("SCOOTER")
    SCOOTER,
    @SerialName("CROUS_AND_GO")
    CROUS_AND_GO,
    @SerialName("SANDWICH_SHOP")
    SANDWICH_SHOP
}
